### PR TITLE
fix: fork nested hp [DET-5945]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

Add support for reverting flattened nested trial hyperparameters to experiment config hyperparameters.

## Test Plan

- [ ] 1. Run a multi-trial experiment with nested hp. 2. Continue trial for one of the trials. Confirm that the continued trial experiment has a valid nested hp in experiment config.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/utils/experiment.ts [hunk 1]
```diff
@@ -23,3 +24,26 @@ export const isSingleTrialExperiment = (experiment: ExperimentBase): boolean =>
   return experiment?.config.searcher.name === ExperimentSearcherName.Single
         || experiment?.config.searcher.max_trials === 1;
 };
+
+export const trialHParamsToExperimentHParams = (
```
Comment (hamidzr) [inline webui/react/src/utils/experiment.ts:28]: suggestion: I'd name the part that's unflattening the dictionary something more generic and separate from trialparams and give it a comment. maybe make `.` a paramter to it.

**Review by hamidzr — APPROVED** [review]
LGTM.  thanks for adding the unit tests as well

File: webui/react/src/utils/experiment.ts (hunk 1 above)
Comment (hamidzr) [reply webui/react/src/utils/experiment.ts:28]: I can't remember does the flattening of the hparams happen anywhere in our code? if it did using this we could add a set of experiment configs with nested hyperparams to our fixtures and make sure that`forEach config c => unflatten(flatten(c)) === c`

**Review by hamidzr — APPROVED** [review]
lgtm. thanks for adding the unit tests

File: webui/react/src/utils/experiment.ts (hunk 1 above)
Comment (hkang1) [reply webui/react/src/utils/experiment.ts:28]: good thing you mentioned this, totally forgot about `flattenObject` in `utils/data.ts`!

Comment (hamidzr) [reply webui/react/src/utils/experiment.ts:12]: looks good : )

**Pull request merged**